ui/app/utils/default_jobs/service-discovery.js

```diff
@@ -0,0 +1,41 @@
+export default `job "service-discovery-example" {
+  datacenters = ["dc1"]
+
+  group "cache" {
```

Comment:
IMO this one should have two groups with a task each, and they'll talk to eachother.

Maybe the "outer" task just curls with exec and the inner is a busybox with some version hello world (like below)?

Reply:
Yeah, it would be nice to have two groups. Here's an example:

```hcl
job "service-discovery-example" {
  datacenters = ["dc1"]

  group "client" {
    task "curl" {
      driver = "docker"

      config {
        image   = "curlimages/curl:7.87.0"
        command = "/bin/ash"
        args    = ["local/script.sh"]
      }

      template {
        data        = <<EOF
#!/usr/bin/env ash

while true; do
{{range nomadService "nomad-service-discovery-example-server"}}
  curl -L -v http://{{.Address}}:{{.Port}}/
{{end}}
  sleep 3
done
EOF
        destination = "local/script.sh"
      }

      resources {
        cpu    = 10
        memory = 50
      }
    }
  }

  group "server" {
    network {
      port "www" {
        to = 8001
      }
    }

    service {
      name     = "nomad-service-discovery-example-server"
      provider = "nomad"
      port     = "www"

      check {
        type     = "http"
        path     = "/"
        interval = "5s"
        timeout  = "1s"
      }
    }

    task "http" {
      driver = "docker"

      config {
        image   = "busybox:1"
        command = "httpd"
        args    = ["-v", "-f", "-p", "${NOMAD_PORT_www}", "-h", "/local"]
        ports   = ["www"]
      }

      template {
        data        = <<EOF
hello world
EOF
        destination = "local/index.html"
      }

      resources {
        cpu    = 10
        memory = 50
      }
    }
  }
}
```

Feel free to put something more meaningful instead of `hello world` 😅 

I also reduced the values in `resource` since these are very small containers.